### PR TITLE
ci: let Netlify post preview URL again

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

**Comment by maelle** [comment]
Now the URL is posted, yay, but I still see an error in the checks log:

```
RequestError [HttpError]: Resource not accessible by integration
    at /home/runner/work/_actions/nwtgck/actions-netlify/v2.0/dist/index.js:10385:21
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  status: 403,
  response: {
    url: 'https://api.github.com/repos/r-lib/pkgdown/deployments',
    status: 403,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      connection: 'close',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Tue, 23 Apr 2024 07:58:25 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-accepted-github-permissions': 'deployments=write',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-api-version-selected': '2022-11-28',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '3480:28A43D:BBA300:1390E52:66276A21',
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4999',
      'x-ratelimit-reset': '1713862705',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '1',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Resource not accessible by integration',
      documentation_url: 'https://docs.github.com/rest/deployments/deployments#create-a-deployment'
    }
  },
  request: {
    method: 'POST',
    url: 'https://api.github.com/repos/r-lib/pkgdown/deployments',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-core.js/3.6.0 Node.js/16.20.2 (linux; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"auto_merge":false,"ref":"7b21d68b32f1286d6f987e89d0ddc96974a2af0a","environment":"pull request","required_contexts":[]}',
    request: { agent: [Agent], hook: [Function: bound bound register] }
  }
}

```

**Comment by maelle** [comment]
@hadley any idea what could be wrong here?

**Comment by hadley** [comment]
To me, it sounds like the resource is not accessible by the integration so you should make it accessible 😆

File: .github/workflows/netlify.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        uses: nwtgck/actions-netlify@v3.0
+        with:
+          publish-dir: './docs'
+          production-branch: main
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          deploy-message:
+            'Deploy from GHA: ${{ github.event.pull_request.title || github.event.head_commit.message }} (${{ github.sha }})'
+          # these default to 'true'
+          enable-commit-comment: false
+          enable-github-deployment: false
```
Comment (hadley) [inline .github/workflows/netlify.yaml:50]: I think this is the only change that's only significant — I don't know the netlify action started doing this by default, but we don't need it and GHA don't have permission for it by default.

**Comment by hadley** [comment]
I pulled the netlify stuff out into a separate file since I think this will make maintenance since it'll no longer get clobbered when we update the standard pkgdown workflow.

**Comment by hadley** [comment]
I also switched the comment back on, since I find that easier to locate.

**Comment by maelle** [comment]
Cool, thanks!

I made a commit mainly to check the action will update its comment as opposed to creating a new comment. :eyes:

**Comment by maelle** [comment]
The docs state "Overwrites comment on pull request (default: true)" so all good :smile_cat:

**Comment by maelle** [comment]
Fine to merge I suppose.

**Comment by hadley** [comment]
I feel like I'm going to get annoyed by the comment thing again, but we'll see!

**Pull request merged**